.github/workflows/build.yml

```diff
@@ -17,44 +20,32 @@ jobs:
     - name: Checkout
       run: git fetch --prune --unshallow --tags
 
-    - name: Set up Go
-      uses: actions/setup-go@v2
-      with:
-        go-version: 1.13
+    - name: Pull compiler image
+      run: docker pull $COMPILER_IMAGE
 
-    - name: Set up Node
-      uses: actions/setup-node@v2
-      with:
-        node-version: '12'
-    
     - name: Cache node modules
       uses: actions/cache@v2
       env:
         cache-name: cache-node_modules
       with:
         path: ui/v2.5/node_modules
-        key: ${{ runner.os }}-build-${{ env.cache-name }}-${{ hashFiles('**/package-lock.json') }}
-        restore-keys: |
-          ${{ runner.os }}-build-${{ env.cache-name }}-
-          ${{ runner.os }}-build-
-          ${{ runner.os }}-
+        key: ${{ runner.os }}-build-${{ env.cache-name }}-${{ hashFiles('ui/v2.5/yarn.lock') }}
 
     - name: Pre-install
-      run: make pre-ui
+      run: docker run --rm --mount type=bind,source="$(pwd)",target=/stash,consistency=delegated -w /stash $COMPILER_IMAGE /bin/bash -c "make pre-ui"
```

Comment:
Please add the following step prior to this:
```
    - name: Start build container
      run: docker run -d --name build --mount type=bind,source="$(pwd)",target=/stash,consistency=delegated -w /stash $COMPILER_IMAGE tail -f /dev/null
```

Change `run` to the following: 
```
run: docker exec -t build /bin/bash -c "make pre-ui"
```

Likewise, change all subsequent `docker run` to `docker exec -t build ...`